src/app/component/layout/components/search-all-results/search-all-results.component.ts

```diff
@@ -40,35 +40,30 @@ export class SearchAllResultsComponent implements OnInit, OnDestroy {
     if (event[VALUE].length > 0) {
       this.inputValue = event[VALUE];
       this.searchSubscription = this.search.getAllSearch(this.inputValue, this.inputValues[0])
-        .subscribe(this.getSearchData.bind(this));
+        .subscribe(data => this.getSearchData(data));
     } else {
       this.resetData();
     }
   }
 
   private getSearchData(data: SearchModel): void {
     this.getNews(data.ecoNews);
-    // tslint:disable-next-line:no-unused-expression
-    data.countOfResults ? this.itemsFound = data.countOfResults : null;
+    this.itemsFound = data.countOfResults ? data.countOfResults : null;
     this.spliceResults();
   }
 
   private getNews(news): void {
-    (news && news.length > 0) ? (this.isSearchFound = true, this.elements = news) : (this.isSearchFound = false);
+    this.isSearchFound = news && news.length;
+    this.elements = this.isSearchFound ? news : this.elements;
   }
 
   private spliceResults(): void {
     const splicedData = this.elements.splice(0, 9);
-    for (let i = 0; i < 9; i++) {
-      if (splicedData[i]) {
-        this.displayedElements.push(splicedData[i]);
-      }
-    }
+    this.displayedElements = [...splicedData.map(elem => elem)];
```

Comment:
Firstly - `map` returns new instance of array, so - you shouldn't use `spread` operations `...`, it's unneeded . Just
```js
this.displayedElements = splicedData.map(elem => elem)
```
Secondly - your changes don't quite correspond to the original code. There will be pushed to the array only if element of `spliceData` is true.
So, change the `map` method in your code on the `filter`.

Reply:
Fixed